17318 error message on invalid parent

hschink opened the pull request:
https://community.openproject.org/work_packages/17318

Commit:
Test error response for invalid children

Commit:
Validate WP children

Originially, this method was defined in a WP patch in OpenProject Backlogs.

TeatroIO commented:
I've prepared a stage. <a href='http://fix-17318-error-message-on-invalid-parent.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

Commit:
Move WP move test to spec and update its specification

With the additional validation of a WP's children, a move of a WP to another
project is only valid if cross project relations are allowed. The old test
didn't consider that setting and just assumed that a WP move is valid across
project borders in any case (even if cross project relations are forbidden).

Commit:
Adapt cukes to child WP validation
